public/app/features/query/state/PanelQueryRunner.ts

```diff
@@ -314,8 +319,30 @@ export class PanelQueryRunner {
     }
 
     this.subscription = panelData.subscribe({
-      next: (data) => {
-        this.lastResult = skipPreProcess ? data : preProcessPanelData(data, this.lastResult);
+      next: async (data) => {
+        const panelDataProcessed = preProcessPanelData(data, this.lastResult);
+
+        //--- check if alert state history uses Loki as implementation, if so, fetch data from Loki state history and concat it to annotations
+        const historyImplementation = getHistoryImplementation();
+        const usingLokiAsImplementation = historyImplementation === StateHistoryImplementation.Loki;
```

Comment:
How does it work for implementations other than `Loki`? It would be good to keep all the code responsible for fetching alert state history in a single place

Reply:
in case we don't use Loki implementation we don't fetch the ash new endpoint. Agreed in moving the code 👍